src/urllib3/request.py

```diff
@@ -84,7 +88,20 @@ def request(
 
         urlopen_kw["request_url"] = url
 
+        if json is not None and body is not None:
+            raise TypeError(
+                "request got values for both 'json' and 'body', can only specify one. Prefer json"
+            )
+
+        if json is not None:
+            if headers is None:
+                headers = self.headers.setdefault("Content-Type", "application/json")
+            # orjson is strict UTF-8
+            body = orjson.dumps(json)
+
         if body is not None:
+            # Remove spaces for delimiters after encoding
+            body = re.sub(b"[^\\S\r\n\t]", b"", body)
```

Comment:
You don't need a regex here, just pass `delimiters=(',', ':')` to `json.dumps` (documented here: https://docs.python.org/3/library/json.html#json.dump)

Reply:
(And this should only be done in the `json` case, there's no reason to touch an existing `body`)

Reply:
Fixed this. Thanks
Regex is always fun!